package NYT.NObjectClient.NProto;

import "yt/core/misc/guid.proto";

////////////////////////////////////////////////////////////////////////////////

message TPrerequisiteTransaction
{
    required NYT.NProto.TGuid transaction_id = 1;
}

message TPrerequisiteRevision
{
    required string path = 1;
    required NYT.NProto.TGuid transaction_id = 2;
    required int64 revision = 3;
}

message TReqExecute
{
    // Each Execute call carries a number of YPath requests.
    // Each of these requests is represented by a message.
    // Each message, in turn, is a sequence of parts.
    // This array contains the lengths of those part sequences.
    // In particular, its length is just the number of encaplusated YPath requests.
    repeated int32 part_counts = 1;

    // Describes transactions that must be alive in order for the request to be executed.
    repeated TPrerequisiteTransaction prerequisite_transactions = 2;

    // Describes node revisions that must match in order for the request to be executed.
    repeated TPrerequisiteRevision prerequisite_revisions = 3;
}

message TRspExecute
{
    // Similar to TReqExecute:::part_counts, but denotes the number of parts
    // in each encapsulated response.
    repeated int32 part_counts = 1;
}

////////////////////////////////////////////////////////////////////////////////

message TReqGCCollect
{ }

message TRspGCCollect
{ }

////////////////////////////////////////////////////////////////////////////////
